ui/src/app/machines/views/MachineDetails/MachineLogs/DownloadMenu/DownloadMenu.tsx

```diff
@@ -1,19 +1,80 @@
+import { useContext, useEffect, useRef } from "react";
+
 import { ContextualMenu } from "@canonical/react-components";
+import { nanoid } from "@reduxjs/toolkit";
 import { format } from "date-fns";
 import fileDownload from "js-file-download";
-import { useSelector } from "react-redux";
+import { useDispatch, useSelector } from "react-redux";
 
 import { useGetInstallationOutput } from "../hooks";
 
+import FileContext from "app/base/file-context";
+import { actions as machineActions } from "app/store/machine";
 import machineSelectors from "app/store/machine/selectors";
 import type { Machine } from "app/store/machine/types";
 import type { RootState } from "app/store/root/types";
-type Props = { systemId: Machine["system_id"] };
+type Props = {
+  systemId: Machine["system_id"];
+};
 export const DownloadMenu = ({ systemId }: Props): JSX.Element | null => {
+  const dispatch = useDispatch();
   const machine = useSelector((state: RootState) =>
     machineSelectors.getById(state, systemId)
   );
   const installationOutput = useGetInstallationOutput(systemId);
+  const getSummaryXmlKey = useRef(nanoid());
+  const getSummaryYamlKey = useRef(nanoid());
+  const fileContext = useContext(FileContext);
+  const summaryXML = fileContext.get(getSummaryXmlKey.current);
+  const summaryYAML = fileContext.get(getSummaryYamlKey.current);
+  const today = format(new Date(), "yyyy-MM-dd");
+  const toggleDisabled =
+    !installationOutput?.log && !summaryYAML && !summaryXML;
+
+  useEffect(() => {
+    if (machine) {
+      // Request the files for this
+      dispatch(
+        machineActions.getSummaryXml(systemId, getSummaryXmlKey.current)
+      );
+      dispatch(
+        machineActions.getSummaryYaml(systemId, getSummaryYamlKey.current)
+      );
+    }
+  }, [dispatch, systemId, machine]);
+
+  // Clean up the requested files when the component unmounts.
+  useEffect(
+    () => () => {
+      fileContext.remove(getSummaryXmlKey.current);
+      fileContext.remove(getSummaryYamlKey.current);
+    },
+    [fileContext]
+  );
+
+  const generateItem = (
+    title: string,
+    filename: string,
+    extension: string,
+    fileContent?: string | null
+  ) =>
+    fileContent
```

Comment:
This is a bit long to be a ternary ;)

up2u

Reply:
Thanks, added an old-school `if`.